frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -1242,6 +1428,13 @@ impl<T: Config> Pallet<T> {
 	}
 
 	fn do_elect() -> Result<(Supports<T::AccountId>, Weight), ElectionError> {
+		// We have to unconditionally try finalizing the signed phase here. There are only two
+		// possibilities:
+		//
+		// - signed phase was open, in which case this is essential for correct functioning of the system
+		// - signed phase was complete or not started, in which case finalization is idempotent and
```

Comment:
fancy adding a test case for this assumption? that `in which case finalization is idempotent `

Reply:
3325691f0